plaza/routes/files/files.go

```diff
@@ -45,21 +45,29 @@ type file_t struct {
 }
 
 func Post(w http.ResponseWriter, r *http.Request) {
-	var filePath string
+	var dst *os.File
 	username := r.URL.Query()["username"][0]
 	filename := r.URL.Query()["filename"][0]
-
-	if runtime.GOOS == "windows" {
-		home := `C:\Users\` + username + `\Desktop\Nanocloud`
-		_, err := os.Stat(home)
-		if os.IsNotExist(err) {
-			os.MkdirAll(home, 0777)
+	path := fmt.Sprintf("/home/%s/%s", username, filename)
+
+	_, err := os.Stat(path)
+	// if a file with exactly the same name already exists
+	if err == nil {
+		// we rename it like 'file (2).txt'
+		for i := 1; i > 0; i++ {
+			extension := filepath.Ext(path)
+			new_file := path[0:len(path)-len(extension)] + " (" + strconv.Itoa(i) + ")" + extension
+			_, err = os.Stat(new_file)
+			if err != nil {
+				dst, err = os.Create(new_file)
+				if err == nil {
+					break
+				}
```

Comment:
Can you handle the case where err != nil here ?

If filesystem is in error state, you got an infinite loop, you should log the error and break whether there is or isn't an error here